controllers/storagecluster/reconcile.go

```diff
@@ -355,9 +355,24 @@ func (r *StorageClusterReconciler) reconcilePhases(
 
 	// Check GetDeletionTimestamp to determine if the object is under deletion
 	if instance.GetDeletionTimestamp().IsZero() {
+		updateRequired := false
 		if !contains(instance.GetFinalizers(), storageClusterFinalizer) {
 			r.Log.Info("Finalizer not found for StorageCluster. Adding finalizer.", "StorageCluster", klog.KRef(instance.Namespace, instance.Name))
 			instance.ObjectMeta.Finalizers = append(instance.ObjectMeta.Finalizers, storageClusterFinalizer)
+			updateRequired = true
+		}
+
+		if backwardCompatibilityMode, err := r.calculateBackwardCompatibilityMode(instance); err != nil {
+			return reconcile.Result{}, err
+		} else if backwardCompatibilityMode != nil {
+			updateRequired = util.AddAnnotation(
+				instance,
+				util.BackwardCompatabilityInfoAnnotationKey,
+				string(util.JsonMustMarshal(&backwardCompatibilityMode)),
```

Comment:
a bit confused, weren't we already adding the same key w/ a different value in upgrade controller for tracking internal consumer id in provider mode?